extensions/amp-intersection-observer-polyfill/0.1/amp-intersection-observer-polyfill.js

```diff
@@ -27,7 +28,9 @@ const TAG = 'amp-intersection-observer-polyfill';
 function upgrade(win) {
   upgradePolyfill(win, () => {
     installIntersectionObserver();
-    // TODO(dvoytenko): hookup 3p host updater for inabox.
+    if (win != win.top) {
```

Comment:
I have a question about `win.top`. I know the AMP doc can be served from a full screen iframe. Will that affect the check here?

Reply:
No, should not have any effect on fullscreen mode. This check overall not even necessary. But I felt like it reads cleaner: "do this only when iframed".

Reply:
I see. a bit confusing though, saw Justin had the question : P

Reply:
Yep. 2 out of 3 people got confused - it's definitely a signal to rework the code. I will do so.

Reply:
Reworked. I think it's a bit better now. PTAL.